File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        
        if (root == null){
            return root;
        } else {
            TreeNode i = root;
            TreeNode left = root.left;
            TreeNode right = root.right;
            i.left = right;
            i.right = left;
            invertTree(left);
            invertTree(right);
            return root;
        }
        
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        #Idea, bfs choosing right most element

        q = deque([root])
        res = []
        while q:
            qlen = len(q)
            for i in range(qlen):
                elem = q.popleft()
                if elem:
                    if elem.left:
                        q.append(elem.left)
                    if elem.right:
                        q.append(elem.right)
                if i == qlen - 1 and elem:
                    res.append(elem.val)
        return res
